cataclysmbnteam/Cataclysm-BN
# Display fridge buff in `item::info`

#### Summary



SUMMARY: Interface "Display fridge/freezer buff to rot time in `item::info`"

#### Purpose of change

Close #1443



#### Describe the solution

* Refactored `item::info` to return the vector that it had as one of the arguments, rather than a string based on it - it seems to be used more than the string it returned before
* Created `item::info_string` that returns said string
* Added a new argument to `item::info` - a `temperature_flag` (as a simple enum right now, we may want a struct if we get more of those "context arguments", like "it is wielded by this NPC, so also print statblock modified by their stats")
* Remove sneaky +5 F temperature penalty for items carried on person
* Use `units::temperature` in few more places
* [x] Supply the new argument in all relevant places
* [x] Figure out how should the UI entry look like: proper "at current temp, will rot in 3.123 days", just "will last at least 2 days in current conditions"?
  * Currently: "It is/isn't protected from rot in current storage. It will last at least [time it will last at highest possible temperature this flag allows]"
* [x] Get #2206 working and merged, since it's a bit pointless to display the buff when there are bugs that wreck it



#### Describe alternatives you've considered

* Bringing back the old "cold" and "frozen" states - would require bringing back item temperature, which is a massive overkill for something as simple as showing rot times
* Hack "cold" and "frozen" states in some different way - might even be easier than this PR and even solve more cases, but could also have harder to find edge cases



#### Testing

* Spawn food
* Spawn fridge, battery, solar panels
* Put food in fridge
* Check the item's description with the fridge off
  * Should say that the item is not protected from rot
* Check the item's description with the fridge working
  * Should say that the item is protected from rot and should show guaranteed freshness time about 385% of shelf time



#### Additional context
